src/chainbin/nnet3-chain-combine.cc

```diff
@@ -44,7 +96,6 @@ int main(int argc, char *argv[]) {
     bool batchnorm_test_mode = false,
         dropout_test_mode = true;
```

Comment:
There are situations where the number of models supplied is quite large (several hundred) and it would be too slow to test every single additional model.  I think you should add an option --max-objective-evaluations, defaulting to 30, and you'll use it to work out a number num_to_add = (num_models + max_objective_evaluations - 1) / max_objective_evaluations, which is >= 1.  Then each time before you re-evaluate the objective function, you'll add num_to_add models to the moving average.  This will keep it fast in larger-scale setups.

It will require that you change your interface- you'll have to split  UpdateNnetMovingAverageAndComputeObjf() into two functions.  Anyway, it wasn't a very pretty interface.